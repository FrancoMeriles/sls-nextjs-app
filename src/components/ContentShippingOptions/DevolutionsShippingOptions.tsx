import {
  Box,
  Heading,
  Button,
  Flex,
  Text,
  Modal,
  ModalOverlay,
  ModalContent,
  ModalHeader,
  ModalBody,
  ModalCloseButton,
  useDisclosure,
  ModalFooter,
} from '@chakra-ui/react'

// Icons
import { Devolutions, Money, Calendar } from '@base/icons'

const DevolutionsShippingOptions = () => {
  const { onOpen, onClose, isOpen } = useDisclosure()

  return (
    <>
      <Flex m="15px 0">
        <Box w="30px" d="flex" alignItems="top" pt={1}>
          <Devolutions color="brand.200" />
        </Box>
        <Box>
          <Text color="brand.200" fontSize="sm">
            Devolución gratis
          </Text>
          <Text fontSize="xs" color="gray.500">
            Tenés 30 días desde que lo recibís.
          </Text>

          <Button onClick={onOpen} size="xs" colorScheme="brand" variant="link" mt={2}>
            Conocer más
          </Button>
        </Box>
      </Flex>
      <Modal isOpen={isOpen} onClose={onClose} size="4xl" isCentered>
        <ModalOverlay />
        <ModalContent padding={6}>
          <ModalHeader>
            <Heading as="h3" size="lg">
              Podés devolver este producto gratis. ¡No importa el motivo!
            </Heading>
          </ModalHeader>
          <ModalCloseButton color="brand.300" />
          <ModalBody p={8}>
            <Flex marginTop={4}>
              <Flex w="60px" alignItems="center">
                <Calendar boxSize={7} />
              </Flex>
              <Box>
                <Text fontSize="md">Tenés 30 días desde que te llega.</Text>
              </Box>
            </Flex>
            <Flex marginTop={8}>
              <Flex w="60px" alignItems="center">
                <Devolutions boxSize={7} />
              </Flex>
              <Box>
                <Text fontSize="md">Podés enviarlo desde tu correo más cercano.</Text>
              </Box>
            </Flex>
            <Flex marginTop={8}>
              <Flex w="60px" alignItems="center">
                <Money boxSize={7} />
              </Flex>
              <Box>
                <Text fontSize="md">
                  Para devolverlo, el producto debe estar tal cual lo recibiste.
                </Text>
              </Box>
            </Flex>
          </ModalBody>
          <ModalFooter>
            <Button variant="ghost" onClick={onClose}>
              Cerrar
            </Button>
          </ModalFooter>
        </ModalContent>
      </Modal>
    </>
  )
}

export default DevolutionsShippingOptions
