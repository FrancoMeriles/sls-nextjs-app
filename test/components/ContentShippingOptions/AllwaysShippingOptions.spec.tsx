import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import userEvent from '@testing-library/user-event'

import DevolutionsShippingOptions from '@base/components/ContentShippingOptions/DevolutionsShippingOptions'
import { useDisclosure } from '@chakra-ui/react'

jest.mock('@chakra-ui/react', () => {
  const originalModule = jest.requireActual('@chakra-ui/react')
  return {
    __esModule: true,
    ...originalModule,
    useDisclosure: jest.fn(),
  }
})

const mockModalDevolutiongOnOpen = jest.fn()
const mockModalDevolutiongIsOpen = false
const mockModalDevolutiongOnClose = jest.fn()

const useDisclosureMock = useDisclosure as jest.Mock

useDisclosureMock.mockImplementation(() => ({
  onOpen: mockModalDevolutiongOnOpen,
  isOpen: mockModalDevolutiongIsOpen,
  onClose: mockModalDevolutiongOnClose,
}))

describe('DevolutionShippingOptions', () => {
  it('should render and match snapshot with required props', () => {
    const tree = renderer.create(<DevolutionsShippingOptions />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should renders and execute open model when user click the button', async () => {
    render(<DevolutionsShippingOptions />)
    expect(screen.getByText('Devolución gratis')).toBeInTheDocument()
    expect(screen.getByText('Tenés 30 días desde que lo recibís.')).toBeInTheDocument()

    userEvent.click(screen.getByText('Conocer más'))
    expect(mockModalDevolutiongOnOpen).toHaveBeenCalled()
  })
})
