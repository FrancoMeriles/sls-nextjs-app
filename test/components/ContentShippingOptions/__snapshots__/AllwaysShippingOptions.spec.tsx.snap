// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevolutionShippingOptions should render and match snapshot with required props 1`] = `
<div
  className="css-1kowmw4"
>
  <div
    className="css-1pla5ih"
  >
    <svg
      className="chakra-icon css-me87sc"
      focusable={false}
      viewBox="0 0 14 12"
    >
      <path
        d="M2.474 7.2h7.225a2.7 2.7 0 1 0 0-5.4H7V.6h2.7a3.9 3.9 0 1 1 0 7.8H2.473l2.45 2.389-.839.859L.14 7.8l3.945-3.848.838.859L2.473 7.2z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="css-0"
  >
    <p
      className="chakra-text css-10wnejt"
    >
      Devolución gratis
    </p>
    <p
      className="chakra-text css-1bmfkgj"
    >
      Tenés 30 días desde que lo recibís.
    </p>
    <button
      className="chakra-button css-1wd94o"
      onClick={[MockFunction]}
      type="button"
    >
      Conocer más
    </button>
  </div>
</div>
`;
